staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -258,3 +290,158 @@ func (c *CacheDelegator) ReadinessCheck() error {
 func (c *CacheDelegator) RequestWatchProgress(ctx context.Context) error {
 	return c.storage.RequestWatchProgress(ctx)
 }
+
+func (c *CacheDelegator) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopCh)
+	})
+	c.wg.Wait()
+}
+
+func newConsistencyChecker(resourcePrefix string, newListFunc func() runtime.Object, cacher storeCache, etcd store, clock clock.Clock) *consistencyChecker {
+	return &consistencyChecker{
+		resourcePrefix: resourcePrefix,
+		newListFunc:    newListFunc,
+		cacher:         cacher,
+		etcd:           etcd,
+		clock:          clock,
+	}
+}
+
+type consistencyChecker struct {
+	resourcePrefix string
+	newListFunc    func() runtime.Object
+
+	cacher storeCache
+	etcd   store
+	clock  clock.Clock
+}
+
+type storeCache interface {
+	store
+	Ready() bool
+}
+
+type store interface {
+	GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error
+}
+
+func (c consistencyChecker) startChecking(stopCh <-chan struct{}) {
+	delay := ConsistencyCheckDelay + time.Duration(rand.Int63n(int64(ConsistencyCheckPeriod-ConsistencyCheckDelay)))
+	select {
+	case <-stopCh:
+		return
+	case <-c.clock.After(delay):
+	}
+	ctx := wait.ContextForChannel(stopCh)
+	for {
+		c.check(ctx)
+		select {
+		case <-stopCh:
+			return
+		case <-c.clock.After(ConsistencyCheckPeriod):
+		}
+	}
+}
+
+func (c *consistencyChecker) check(ctx context.Context) {
+	digests, err := c.calculateDigests(ctx)
+	if err != nil {
+		klog.ErrorS(err, "Cache consistentency check error")
+		return
+	}
+	if digests.CacheDigest == digests.EtcdDigest {
+		klog.V(3).InfoS("Cache consistentency check passed", "resourceVersion", digests.ResourceVersion, "digest", digests.CacheDigest)
+	} else {
+		klog.InfoS("Cache consistentency check failed", "resourceVersion", digests.ResourceVersion, "etcdDigest", digests.EtcdDigest, "cacheDigest", digests.CacheDigest)
```

Comment:
It probably should be more than just info log :)

Reply:
Done